roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -33,12 +33,23 @@ class DockerImageAvailability(DockerHostMixin, OpenShiftCheck):
     # we use python-docker-py to check local docker for images, and skopeo
     # to look for images available remotely without waiting to pull them.
     dependencies = ["python-docker-py", "skopeo"]
-    skopeo_img_check_command = "timeout 10 skopeo inspect --tls-verify=false docker://{registry}/{image}"
 
     def __init__(self, *args, **kwargs):
         super(DockerImageAvailability, self).__init__(*args, **kwargs)
         # record whether we could reach a registry or not (and remember results)
         self.reachable_registries = {}
+        self.skopeo_img_check_command = "timeout 10 skopeo inspect --tls-verify=false docker://{registry}/{image}"
+        self.oreg_auth_user = None
+        self.oreg_auth_password = None
+
+        oreg_auth_user = self.get_var('oreg_auth_user', default='')
+        oreg_auth_password = self.get_var('oreg_auth_password', default='')
+        if oreg_auth_user != '' and oreg_auth_password != '':
+            # all the options must be grouped together or skopeo will fail
+            self.skopeo_img_check_command = "timeout 10 skopeo inspect --tls-verify=false"\
+                                            " --creds='{oau}:{oap}' docker://{registry}/{image}"
```

Comment:
Come to think of it, since the `command` module doesn't run through shell, I'm not sure the single quotes are needed; they may even be considered part of the arg which would break things. The command module docs don't exactly say... @michaelgugino have you had a chance to test against an authenticated registry?

Reply:
I have tested locally with quotes on the cli using skopeo, I haven't tested the command module locally.

I can try that, but my password doesn't include any special characters (it's a token generated and is alpha-numeric).

Reply:
Simple testing indicates quotes do what you would expect for arguments, aren't treated as part of them. However it would still be an issue if the contents have single quotes.